x-pack/plugins/remote_clusters/public/shim.js

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { i18n } from '@kbn/i18n';
+import { FormattedMessage } from '@kbn/i18n/react';
+import { I18nContext } from 'ui/i18n';
+import chrome from 'ui/chrome';
+import { DOC_LINK_VERSION, ELASTIC_WEBSITE_URL } from 'ui/documentation_links';
+import { management, MANAGEMENT_BREADCRUMB } from 'ui/management';
+import { fatalError, toastNotifications } from 'ui/notify';
+import routes from 'ui/routes';
+
+import { trackUiMetric as track } from '../../../../src/legacy/core_plugins/ui_metric/public';
+
+export function createShim() {
+  return {
+    core: {
+      translation: {
+        i18n,
+        Context: I18nContext,
+        FormattedMessage,
+      },
+      routing: {
+        registerAngularRoute: (path, config) => {
+          routes.when(path, config);
+        },
+      },
+      chrome,
+      notification: {
+        fatalError,
+        toastNotifications,
+      },
+      documentation: {
+        esDocBasePath: `${ELASTIC_WEBSITE_URL}guide/en/elasticsearch/reference/${DOC_LINK_VERSION}/`,
+      },
```

Comment:
The shape of this is a pretty different than CoreStart. It's definitely helpful to use TypeScript here to have it guide the correct shape.

This might be a bit closer:
```js
return {
  core: {
    i18n: { Context: I18nContext },
    chrome: {
      // Only include chrome APIs we actually support in CoreStart.
      // It's actually like that you don't need chrome at all and instead need `http.basePath` for example.
    },
    notification: {
      toast: toastNotifications, // I'm not sure, but I think this API is actually slightly different from CoreStart.
    },
    http: {
      basePath: { prepend }
    }
  }
}
```

You'll notice I removed some things that are not present CoreStart (and likely won't ever be). I think it's best to do this now (or in a subsequent PR if you prefer) so that you can start shaping your application the correct way.

For instance, the `FormattedMessage` component should be imported directly from @kbn/i18n rather than injected via core contracts. Another thing that core will not support is routing. We recommend you setup ReactRouter similar to how APM is doing this.

If you want to do this more incrementally, you can start with what you have here and start replacing it with pieces of the real CoreStart instance from `import { npStart: { core } } from 'ui/new_platform'` in future PRs, rather than constructing your own.

It's also worth mentioning that this interface is changing a bit right now. For instance, Chrome and I18n are moving from CoreSetup to CoreStart in #37915